exporter/elasticexporter/internal/translator/elastic/traces.go

```diff
@@ -0,0 +1,467 @@
+			// are expected to send the scheme; this is just
+			// a failsafe.
+			context.model.Request.URL.Protocol = "http"
+		}
+		if context.model.Request.URL.Hostname == "" {
+			context.model.Request.URL.Hostname = netHostName
+		}
+		if context.model.Request.URL.Port == "" && netHostPort > 0 {
+			context.model.Request.URL.Port = strconv.Itoa(netHostPort)
+		}
+		if netPeerIP != "" {
+			remoteAddr := netPeerIP
+			if netPeerPort > 0 {
+				remoteAddr = net.JoinHostPort(remoteAddr, strconv.Itoa(netPeerPort))
+			}
+			context.setHTTPRemoteAddr(remoteAddr)
+		}
+	}
+	return nil
+}
+
+func setSpanProperties(otlpSpan *otlptrace.Span, span *model.Span) error {
+	var (
+		context     spanContext
+		netPeerName string
+		netPeerIP   string
+		netPeerPort int
+	)
+
+	for _, attr := range otlpSpan.GetAttributes() {
+		switch k := attr.GetKey(); k {
+		// http.*
+		case conventions.AttributeHTTPURL:
+			if err := context.setHTTPURL(attr.GetStringValue()); err != nil {
+				return err
+			}
+		case conventions.AttributeHTTPTarget:
+			if err := context.setHTTPURL(attr.GetStringValue()); err != nil {
+				return err
+			}
+		case conventions.AttributeHTTPHost:
+			context.setHTTPHost(attr.GetStringValue())
+		case conventions.AttributeHTTPScheme:
+			context.setHTTPScheme(attr.GetStringValue())
+		case conventions.AttributeHTTPStatusCode:
+			context.setHTTPStatusCode(int(attr.GetIntValue()))
+
+		// net.*
+		case conventions.AttributeNetPeerName:
+			netPeerIP = attr.GetStringValue()
+		case conventions.AttributeNetPeerIP:
+			netPeerIP = attr.GetStringValue()
+		case conventions.AttributeNetPeerPort:
+			netPeerPort = int(attr.GetIntValue())
+
+		// db.*
+		case conventions.AttributeDBType:
+			context.setDatabaseType(attr.GetStringValue())
+		case conventions.AttributeDBInstance:
+			context.setDatabaseInstance(attr.GetStringValue())
+		case conventions.AttributeDBStatement:
+			context.setDatabaseStatement(attr.GetStringValue())
+		case conventions.AttributeDBUser:
+			context.setDatabaseUser(attr.GetStringValue())
+
+		// other: record as a tag
+		default:
+			context.model.Tags = append(context.model.Tags, model.IfaceMapItem{
+				Key:   cleanLabelKey(k),
+				Value: ifaceAttributeValue(attr),
+			})
+		}
+	}
+
+	span.Type = "app"
+	if context.model.HTTP != nil {
+		span.Type = "external"
+		span.Subtype = "http"
+		if context.http.URL != nil {
+			if context.http.URL.Scheme == "" {
+				// A bit presumptuous, but OpenTelemetry clients
+				// are expected to send the scheme; this is just
+				// a failsafe.
+				context.http.URL.Scheme = "http"
+			}
+			if context.http.URL.Host == "" {
+				hostname := netPeerName
+				if hostname == "" {
+					hostname = netPeerIP
+				}
+				if hostname != "" {
+					host := hostname
+					if netPeerPort > 0 {
+						port := strconv.Itoa(netPeerPort)
+						host = net.JoinHostPort(hostname, port)
+					}
+					context.http.URL.Host = host
+				}
+			}
+		}
+	}
+	if context.model.Database != nil {
```

Comment:
Should this be an `else if`?

Reply:
Some databases use HTTP for their transport, so no. I don't know if it'll occur in practise, but I think it's logically sound to have a database span with an HTTP URL.